pyserini/index/pyutils.py

```diff
@@ -87,24 +87,28 @@ def __init__(self, index_dir):
         self.object = JIndexReaderUtils()
         self.reader = self.object.getReader(JString(index_dir))
 
-    def analyze(self, text: str) -> List[str]:
-        """Applies Anserini's default Lucene ``Analyzer`` to process a piece of text.
+    def analyze(self, text: str, analyzer=None) -> List[str]:
+        """Analyzes a piece of text. Applies Anserini's default Lucene analyzer if analyzer not specified.
 
         Parameters
         ----------
         text : str
             The piece of text to analyze.
-
+        analyzer : analyzer
+            The analyzer to apply.
         Returns
         -------
         List[str]
-            List of tokens corresponding to the output of the ``Analyzer``.
+            List of tokens corresponding to the output of the analyzer.
         """
-        stemmed = self.object.analyze(JString(text))
-        token_list = []
-        for token in stemmed.toArray():
-            token_list.append(token)
-        return token_list
+        if analyzer is None:
+            results = JAnalyzerUtils.analyze(JString(text))
```

Comment:
Here and line 107 the `text` should be replace with `text.encode('utf-8')` I think